lib/fieldhand/identify_parser.rb

```diff
@@ -0,0 +1,45 @@
+require 'fieldhand/identify'
+require 'fieldhand/response_parser'
+
+module Fieldhand
+  # A parser for Identify responses.
+  #
+  # See https://www.openarchives.org/OAI/openarchivesprotocol.html#Identify
+  class IdentifyParser
+    attr_reader :response
+
+    # Return a new parser for the given response body.
+    def initialize(response)
+      @response = response
+    end
+
+    # Return an `Enumerator` of `Identify`s for each found in the response.
+    #
+    # In reality, there will only ever be at most one `Identify` in a response but having a consistent interface with
+    # the other parsers keeps the supporting code simpler.
+    #
+    # Raises a `ProtocolError` for any errors found in the response.
+    def items
+      return enum_for(:items) unless block_given?
+
+      response_parser.errors.each do |error|
```

Comment:
I wonder if the repetition of this error-handling logic and the `resumption_token` method below between all parsers mean that we should further separate the base `ResponseParser` and the more specific parsers (e.g. `ListMetadataFormatsParser`).

Perhaps the specific parsers should _only_ handle returning items and we can leave the resumption token and error handling to the response parser.

e.g. some Ruby pseudocode for the coördination:

```ruby
response_parser = ResponseParser.new(request(query.merge('verb' => verb)))
response_parser.errors.each do |error|
  raise error
end

item_parser = parser_class.new(response_parser)
item_parser.items.each do |item|
  yield item
end

break unless response_parser.resumption_token
```